File: back/DevC/src/main/java/com/back/devc/domain/post/post/dto/PostUpdateResponse.java
package com.back.devc.domain.post.post.dto;

import com.back.devc.domain.post.post.entity.Post;

public record PostUpdateResponse(
        Long postId,
        String title,
        String content,
        Long categoryId
) {
    public static PostUpdateResponse from(Post post) {

        var category = post.getCategory();

        return new PostUpdateResponse(
                post.getPostId(),
                post.getTitle(),
                post.getContent(),
                category != null ? category.getCategoryId() : null
        );
    }
}

File: back/DevC/src/main/kotlin/com/back/devc/domain/post/post/dto/PostUpdateResponse.kt
package com.back.devc.domain.post.post.dto

import com.back.devc.domain.post.post.entity.Post

data class PostUpdateResponse(
    val postId: Long,
    val title: String,
    val content: String,
    val categoryId: Long
) {
    companion object {
        @JvmStatic
        fun from(post: Post): PostUpdateResponse {
            return PostUpdateResponse(
                postId = post.postId,
                title = post.title,
                content = post.content,
                categoryId = post.category.categoryId
            )
        }
    }
}